code/game/objects/items/mail.dm

```diff
@@ -332,3 +352,164 @@
+		return ..()
+	else
+		if(IS_WEAKREF_OF(user.mind, made_by_ref))
+			balloon_alert(user, "disarming trap...")
+			if(!do_after(user, 2 SECONDS, target = src))
+				return FALSE
+			balloon_alert(user, "disarmed")
+			playsound(src, 'sound/machines/defib_ready.ogg', vol = 100, vary = TRUE)
+			armed = FALSE
+			return TRUE
+		else
+			balloon_alert(user, "tinkering with something...")
+			
+			if(!do_after(user, 2 SECONDS, target = src))
+				after_unwrap(user)
+				return FALSE
+			if(prob(50))
+				balloon_alert(user, "disarmed something...?")
+				playsound(src, 'sound/machines/defib_ready.ogg', vol = 100, vary = TRUE)
+				armed = FALSE
+				return TRUE
+			else
+				after_unwrap(user)
+				return TRUE
+
+/obj/item/storage/mail_counterfeit_device
+	name = "GLA-2 mail counterfeit device"
+	desc = "Device that actually able to counterfeit NT's mail. This device also able to place a trap inside of mail for malicious actions. Trap will \"activate\" any item inside of mail. Also it might be used for contraband purposes. Integrated micro-computer will give you great configuration optionality for your needs."
+	w_class = WEIGHT_CLASS_NORMAL
+	icon = 'icons/obj/device_syndie.dmi'
+	icon_state = "mail_counterfeit_device"
+
+/obj/item/storage/mail_counterfeit_device/Initialize(mapload)
+	. = ..()
+	atom_storage.max_slots = 1
+	atom_storage.allow_big_nesting = TRUE
+	atom_storage.max_specific_storage = WEIGHT_CLASS_NORMAL
+
+/obj/item/storage/mail_counterfeit_device/examine_more(mob/user)
+	. = ..()
+	. += span_notice("<i>You notice the manufacture marking on the side of the device...</i>")
+	. += "\t[span_info("Guerilla Letter Assembler")]"
+	. += "\t[span_info("GLA Postal Service, right on schedule.")]"
+	return .
+
+/obj/item/storage/mail_counterfeit_device/attack_self(mob/user, modifiers)
+	var/mail_type = tgui_alert(user, "Is it gonna look like an envelope or like a normal mail?", "Mail Counterfeiting", list("Mail", "Envelope"))
+	if(isnull(mail_type))
+		return FALSE
+	if(loc != user)
+		return FALSE
+	mail_type = lowertext(mail_type)
+
+	var/mail_armed = tgui_alert(user, "Is it gonna be armed?", "Mail Counterfeiting", list("Yes", "No"))
+	if(isnull(mail_armed))
+		return FALSE
+	if(loc != user)
+		return FALSE
+
+	var/list/mail_recipients = list("Anyone")
+	var/list/mail_recipients_input_list = list("Anyone")
+	var/list/recipients_dupes = list()
+	for(var/datum/record/locked/person in sort_record(GLOB.manifest.locked))
+		if(isnull(person.mind_ref))
+			continue
+		mail_recipients += person.mind_ref
+		if(recipients_dupes[person.name] == null)
+			if(mail_recipients_input_list.Find(person.name))
+				recipients_dupes[person.name] = 1
+				mail_recipients_input_list += "[person.name] (1)"
+			else
+				mail_recipients_input_list += "[person.name]"
+		else
+			recipients_dupes[person.name] += 1
+			var/i = recipients_dupes[person.name]
+			mail_recipients_input_list += "[person.name] ([i])"
```

Comment:
Once more, CC @MrMelbert what is the proc for doing this so they don't have to? You said heretics use some proc

Reply:
`avoid_assoc_duplicate_keys` is what you want - 
```
///takes an input_key, as text, and the list of keys already used, outputting a replacement key in the format of "[input_key] ([number_of_duplicates])" if it finds a duplicate
///use this for lists of things that might have the same name, like mobs or objects, that you plan on giving to a player as input
```

So all you have to do is have a list for tracking used names
`var/list/used_names = list()`
Then when setting up the recipients list use the proc
`mail_recipients_input_list[avoid_assoc_duplicate_keys(person.name, used_names) = person`
or however the list is formatted

Reply:
i did it a little bit diffirent, but thanks a lot for this cool proc